src/edu/csus/ecs/pc2/exports/ccs/ResultsFile.java

```diff
@@ -292,30 +207,56 @@ public String[] createTSVFileLines(IInternalContest contest, Group group, String
             // 6 Time of the last submission 233 integer
 
             String reservationId = account.getExternalId();
-            boolean ranked = false;
-            if (record.getNumberSolved() >= median) {
-                ranked = true;
+            
+            boolean isHighestHonor = false;
+            boolean isHighHonor = false;
+            boolean isHonor = false;
+
+            if (record.getNumberSolved() > 0) {
+                if (finalizeData.isUseWFGroupRanking()) {
+                    if (record.getNumberSolved() >= highestHonorSolvedCount) {
+                        isHighestHonor = true;
+                    } else if (record.getNumberSolved() >= highHonorSolvedCount) {
+                        isHighHonor = true;
+                    } else if (record.getNumberSolved() >= median) {
+                        isHonor = true;
+                    }
+                }
+                else if (record.getNumberSolved() >= median) {
+                    isHonor = true;
+                }
             }
-            String award = getAwardMedal(record.getRankNumber(), finalizeData, ranked);
+
+            String award = getAwardMedal(record.getRankNumber(), finalizeData, isHighestHonor, isHighHonor, isHonor);
             String rank = "";
             if (!"honorable".equalsIgnoreCase(award)) {
```

Comment:
The method `getAwardMedal()` (called two lines above) does not ever return the literal string "honorable".  It does (sometimes) returns the value of the enum element `ResultsFile.HONORABLE`, which in the current implementation happens to be the string "Honorable".  However, if someone later changes the value of that enum, this test will fail.  The `if` statement should use the enum element, not a fixed string (otherwise, it's defeating the entire purpose of the enum...)

Reply:
This has been in the code since commit [
bug 1156: various fixes to generate a proper results.tsv](https://github.com/SamanwaySadhu/pc2v9/commit/2116715308e93619d1d655a26ee692d4b4f486da) back in June 2017, when it was added.  I would say it is out of scope for this PR, however, if it were to be slipped in as a CI, I wouldn't object.  By the same token, I might prefer an issue be submitted and a separate PR so that it can be fully tested.

Reply:
I'd be ok if this was included in this PR as a "Continuous Improvement".

Reply:
Pushed in the latest commit.